pkg/operator/ceph/cluster/controller.go

```diff
@@ -328,6 +331,14 @@ func (c *ClusterController) configureExternalCephCluster(namespace, name string,
 }
 
 func (c *ClusterController) configureLocalCephCluster(namespace, name string, cluster *cluster, clusterObj *cephv1.CephCluster) error {
+	// When the dataDirHostPath is set, check if the path is in the disallowed paths list
+	cleanDataPath := path.Clean(cluster.Spec.DataDirHostPath)
+	for _, b := range disallowedHostDirectories {
+		if cleanDataPath == b {
+			return fmt.Errorf("dataDirHostPath (given: %s) must not be used", cluster.Spec.DataDirHostPath)
```

Comment:
Perhaps we could also show which directories are not valid?

Reply:
Should all invalid directories be shown or just the conflicting one?